runtime/src/main/cpp/Runtime.cpp

```diff
@@ -140,7 +140,9 @@ RuntimeState* initRuntime() {
 void deinitRuntime(RuntimeState* state, bool destroyRuntime) {
   RuntimeAssert(state->status == RuntimeStatus::kRunning, "Runtime must be in the running state");
   state->status = RuntimeStatus::kDestroying;
-  // This may be called after TLS is zeroed out, so ::memoryState in Memory cannot be trusted.
+  // This may be called after TLS is zeroed out, so ::runtimeState and ::memoryState in Memory cannot be trusted.
+  // TODO: This may in fact reallocate TLS without guarantees that it'll be deallocated again.
+  ::runtimeState = state;
```

Comment:
Zeroing it after `deinitRuntime` in `Kotlin_deinitRuntimeCallback` might also make sense.

Reply:
Good idea, thanks!